pip/index.py

```diff
@@ -413,13 +419,9 @@ def _find_all_versions(self, project_name):
         for location in url_locations:
             logger.debug('* %s', location)
 
-        formats = set(["source"])
-        if self.use_wheel:
-            formats.add("binary")
-        search = Search(
-            project_name.lower(),
-            pkg_resources.safe_name(project_name).lower(),
-            frozenset(formats))
+        canonical_name = pkg_resources.safe_name(project_name).lower()
+        formats = fmt_ctl_formats(self.format_control, canonical_name)
+        search = Search(project_name.lower(), canonical_name, formats)
```

Comment:
This is what the prior code was doing. I think its fine to want to change it but its unrelated to this goal, and since it might cause test fallout etc, I'd rather someone do that separately.